CMakeLists.txt

```diff
@@ -881,6 +881,9 @@ set(UNIT_TEST_OBJ   )
 set(UNIT_TEST_CASES )
 if (COMPILE_TESTS) 
   include(CTest)
+  set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} --coverage)
```

Comment:
Should this be guarded for e.g. GCC/Clang? Or perhaps CMake has some abstraction for this?

Reply:
Soo if enabling this breaks unittesting support for Cygwin, I think it should be guarded in some way.  I'm everything but a cmake expert tho

Reply:
My thinking is that nobody uses Cygwin for development, only occasionally as a "this really has to work on Windows" last-ditch effort, and therefore they don't want/need unit tests. Is this true?

Reply:
Well noone who's a core vampire dev right now but there might be future contributors who want..? I think if we can just add a `if (compiler !=  Cygwin)` guard before that line and everything's fine we might was well do it, right? Or is it more complicated than that?

Reply:
See discussion in #516 - we currently call `_Exit` somewhere because Cygwin didn't like `exit` in 2014 (!), but `_Exit` breaks coverage for some reason. `_Exit` is kind of evil anyway and I'm not sure it breaks Cygwin any more - @inpefess can you confirm or deny?

Reply:
Let me check how it works now with `exit` on Cygwin

Reply:
If you can modify this call to `set()` to something like

```cmake
if(CMAKE_CXX_COMPILER_ID STREQUAL GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang$)
  set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} --coverage)
endif()
```

so that we don't break other compilers in principle, I'm happy to merge. Cygwin may well be broken, but it apparently already was so let's take one step at a time!

Reply:
@MichaelRawson done